kafka/producer/kafka.py

```diff
@@ -551,6 +549,9 @@ def send(self, topic, value=None, key=None, partition=None, timestamp_ms=None):
             value_bytes = self._serialize(
                 self.config['value_serializer'],
                 topic, value)
+            assert type(key_bytes) in (bytes, bytearray, memoryview, type(None))
```

Comment:
@dpkp Say, I don't like the `asserts` in code. Maybe just duck type it to bytes. Ie:
```
key_bytes = bytes(key_bytes)
```